feign/src/test/java/com/baeldung/feign/clients/header/dynamicheader/BookClientLiveTest.java

```diff
@@ -0,0 +1,44 @@
+package com.baeldung.feign.clients.header.dynamicheader;
+
+import com.baeldung.feign.clients.builder.BookFeignClientBuilder;
+import com.baeldung.feign.models.Book;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Consumes https://github.com/Baeldung/spring-hypermedia-api
+ */
+@Slf4j
+public class BookClientLiveTest {
+	
+    private BookFeignClientBuilder feignClientBuilder;
+
+    private BookClient bookClient;
+
+    @Before
+    public void setup() {
+        feignClientBuilder = new BookFeignClientBuilder();
+        bookClient = feignClientBuilder.createClient(BookClient.class, "http://localhost:8081/api/books");
```

Comment:
Can we make this test run against a wiremock or mockserver, so the reader can easily run it themselves. I think it's a bad thing to expect them to start up another example app.

Reply:
i feel the wiremock or mockserver will dilute the example for feign and we might need to explain the wiremock itself. So this code is running with actual endpoint which pretty easy to setup.  Also, previous article has use the same example. So, reusability was also a factor for this article.
But definitely, this is very good point to keep in mind.